assets/my-first-radanalytics-app/sparkpi-java-vertx.adoc

```diff
@@ -0,0 +1,746 @@
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkException;
+import org.apache.spark.api.java.JavaSparkContext;
+....
+
+Next we declare our provider class and set up a few internal variables. The static `INSTANCE` will provide our concrete singular instantiation of this class which defines our singleton. The `sparkConf` and `sparkContext` variables are the actual connections to our Spark cluster.
+
+....
+public class SparkContextProvider {
+
+    private static SparkContextProvider INSTANCE = null;
+
+    private SparkConf sparkConf;
+    private JavaSparkContext sparkContext;
+    private String jarFile;
+....
+
+Since this class will implement the singleton pattern, we make its constructors private to ensure that it will only be instantiated by the `init` method. The second contructor function is the primary method here, it accepts the properties object and instantiates the internal private variables. The `setJars` function will instruct Spark to associate our application Jar with the https://spark.apache.org/docs/latest/api/java/org/apache/spark/SparkConf.html[SparkConf] object, and subsequently the Spark context.
+
+....
+    private SparkContextProvider() {
+    }
+
+    private SparkContextProvider(SparkPiProperties props) {
+        this.sparkConf = new SparkConf().setAppName("JavaSparkPi");
+        this.sparkConf.setJars(new String[]{props.getJarFile()});
+        this.sparkContext = new JavaSparkContext(sparkConf);
+    }
+....
+
+The `init` function is the main entry point for constructing the context provider. This function will simply check to determine if an instance has been created, and if not it will create that instance. As there is always the possibility of failure, this function will also catch any errors that result from spawning the new instance.
+
+....
+    public static boolean init(SparkPiProperties props) {
+        try {
+            if (INSTANCE == null) {
+                INSTANCE = new SparkContextProvider(props);
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            return false;
+        }
+        return true;
+    }
+....
+
+The last function in this class is the primary means of interacting with the context. This function provides a convenient method for any other class to gain the Spark contenxt.
+
+....
+    public static JavaSparkContext getContext() {
+        return INSTANCE.sparkContext;
+    }
+....
+
+== Create the application resource files
+
+In addition to the source files we also need a few resource files to set default properties and configurations for our application. To begin creating your resource files you will first need to make a directory for them by running the following command from the root of your project:
+
+....
+mkdir -p src/main/resources
+....
+
+The first file you will create in that directory is named `application.properties` and it should contain the following contents:
```

Comment:
in the reference repo this file is named `sparkpi.properties`, reverting the filename made my application run. i think this was the issue i ran into.